### PR TITLE
Add support for DeploymentTargetClaim in the Environment controller

### PR DESCRIPTION
**Pull request by chetan-rns**
#### Description:
- Environment reconciler should use the secrets specified in the DeploymentTarget
- Reconciler should update the managed environment details if the DeploymentTarget is updated
- unit tests and e2e tests

#### Link to JIRA Story (if applicable):

https://issues.redhat.com/browse/GITOPSRVCE-388

### CODE REVIEW

**Review by jgwest — CHANGES_REQUESTED** [review]

File: appstudio-controller/controllers/appstudio.redhat.com/environment_controller_test.go [hunk 1]
```diff
@@ -370,5 +372,427 @@ var _ = Describe("Environment controller tests", func() {
...
+			Expect(res).To(Equal(reconcile.Result{}))
+
+			managedEnvCR := managedgitopsv1alpha1.GitOpsDeploymentManagedEnvironment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "managed-environment-" + env.Name,
+					Namespace: req.Namespace,
+				},
+			}
+			err = k8sClient.Get(ctx, client.ObjectKeyFromObject(&managedEnvCR), &managedEnvCR)
+			Expect(err).ToNot(BeNil())
```
[the remark this answers is not in the document]
Comment (chetan-rns) [reply appstudio-controller/controllers/appstudio.redhat.com/environment_controller_test.go:542]: Here we are checking if the resource is not present and the error is not returned from the reconciler. So I suppose we don't have to change this

File: appstudio-controller/controllers/appstudio.redhat.com/environment_controller_test.go [hunk 2]
```diff
@@ -370,5 +372,427 @@ var _ = Describe("Environment controller tests", func() {
...
```
[the remark this answers is not in the document]
Comment (chetan-rns) [reply appstudio-controller/controllers/appstudio.redhat.com/environment_controller_test.go:622]: Here we are checking if the resource is not present and the error is not returned from the reconciler. So I suppose we don't have to change this

File: appstudio-controller/controllers/appstudio.redhat.com/environment_controller.go [hunk 3]
```diff
@@ -241,5 +294,98 @@ func generateEmptyManagedEnvironment(environmentName string, environmentNamespac
 func (r *EnvironmentReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&appstudioshared.Environment{}).
+		Watches(
+			&source.Kind{Type: &appstudioshared.DeploymentTargetClaim{}},
+			handler.EnqueueRequestsFromMapFunc(r.findObjectsForDeploymentTargetClaim),
+			builder.WithPredicates(predicate.GenerationChangedPredicate{}),
```
[the remark this answers is not in the document]
Comment (chetan-rns) [reply appstudio-controller/controllers/appstudio.redhat.com/environment_controller.go:301]: You are right. I just verified and it skipped the events with changes to the `status` field. I've replaced it with `ResourceVersionChangedPredicate`

**Review by jgwest — APPROVED** [review]
Looks great, thanks @chetan-rns!

**Pull request merged**